src/main/java/org/isf/menu/rest/UserController.java

```diff
@@ -123,68 +124,69 @@ public ResponseEntity<List<UserDTO>> getUser(@RequestParam(name = "group_id", re
-			return ResponseEntity.status(HttpStatus.CREATED).body(true);
-		} catch (OHServiceException serviceException) {
-			LOGGER.info("User is not created.");
-			throw new OHAPIException(new OHExceptionMessage("User not created."));
-		}
-	}
-
 	/**
 	 * Updates an existing {@link User}.
 	 * @param userDTO - the {@link User} to update
-	 * @param updatePassword - indicates if it is the password that need to be updated
-	 * @return {@code true} if the user has been updated, {@code false} otherwise.
+	 * @return the updated {@link UserDTO} if the user has been updated.
+	 * @throws OHServiceException throws if the update fails
 	 */
-	@PutMapping(value = "/users", produces = MediaType.APPLICATION_JSON_VALUE)
-	public ResponseEntity<Boolean> updateUser(
-					@Valid @RequestBody UserDTO userDTO,
-					@RequestParam(name = "password", defaultValue = "false") boolean updatePassword) throws OHServiceException {
+	@PutMapping(value = "/users/{username}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<UserDTO> updateUser(
+		@PathVariable("username") String userName,
+		@Valid @RequestBody UserDTO userDTO) throws OHServiceException {
 		String requestUserName = userDTO.getUserName();
-		String currentUser = SecurityContextHolder.getContext().getAuthentication().getName();
-		final String ADMIN = "admin";
-		boolean isAdminOrSameUser = requestUserName.equals(currentUser) || currentUser.equals(ADMIN);
-		boolean isSameUserUpdatingPassword = requestUserName.equals(currentUser) && updatePassword;
-		if (!isAdminOrSameUser || !isSameUserUpdatingPassword) {
-			throw new OHAPIException(new OHExceptionMessage("Not allowed."));
+		if(requestUserName != null && !userName.equals(requestUserName)) {
+			throw new OHAPIException(new OHExceptionMessage("Invalid request payload"));
 		}
-		if (userManager.getUserByName(requestUserName) == null) {
+		if (userManager.getUserByName(userName) == null) {
 			throw new OHAPIException(new OHExceptionMessage("The specified user does not exist."));
 		}
 		User user = userMapper.map2Model(userDTO);
 		boolean isUpdated;
-		if (updatePassword) {
+		if (!user.getPasswd().isEmpty()) {
 			isUpdated = userManager.updatePassword(user);
 		} else {
 			isUpdated = userManager.updateUser(user);
 		}
 		if (isUpdated) {
-			return ResponseEntity.ok(isUpdated);
+			return ResponseEntity.ok(userMapper.map2DTO(userManager.getUserByName(userName)));
 		} else {
 			throw new OHAPIException(new OHExceptionMessage("User not updated."));
 		}
 	}
 
+	/**
+	 * Creates a new {@link User}.
+	 * @param userDTO - the {@link User} to insert
+	 * @return {@code true} if the user has been inserted, {@code false} otherwise.
+	 * @throws OHServiceException 
+	 */
+	@PostMapping(value = "/users", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<Boolean> newUser(@Valid @RequestBody UserDTO userDTO) throws OHServiceException {
+		LOGGER.info("Attempting to create a user.");
```

Comment:
Why isn't the new user name included in the log message?  Seems like it would be helpful.